### PR TITLE
feat(accounts): import account without password

### LINKED ISSUE #18180: [Bug]: Import Account without password
### Describe the bug

not all JSON account files are password protected, currently cannot import an account without a password. Same as this older closed issue.
https://github.com/MetaMask/metamask-extension/issues/5681

### Steps to reproduce

1. Export Account as file, do not password protect
2. Click import account in metamask
3. Select exported file
4. Submit/ import button deactivated until password for file is entered

### Error messages or log output

_No response_

### Version

10.24 +

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by montelaidev**
## Explanation

Makes password optional when importing an account from a JSON file.

* Fixes #18180

### Before
Screenshot 2023-05-15 at 4 49 03 PM

### After

Screenshot 2023-05-17 at 10 06 25 AM

## Manual Testing Steps

1. Click on account icon
2. Click on `Import Account`
3. Select Json for type
4. Click on Password less toggle
5. Upload Json file
6. Click on Import.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [x] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by Nemerem34 — DISMISSED** [review]

**Review by darkwing — DISMISSED** [review]

**Comment by darkwing** [comment]
This area looks really really busy and overwhelming:

SCR-20230515-jfsa

Maybe we can add some spacing?

File: ui/pages/create-account/import-account/json.js [hunk 1]
```diff
@@ -28,18 +35,36 @@ export default function JsonImportSubview({ importAccountFunc }) {
...
+  function isPrimaryDisabled() {
+    if (fileContents === '') {
+      return true;
+    }
+    if (!isPasswordless && password === '') {
+      return true;
+    }
+
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (montelaidev) [reply ui/pages/create-account/import-account/json.js:49]: yes, let me change that.

**Comment by HowardBraham** [comment]
Did this work before if you just left the password field blank?  I'm mostly guessing that because I don't see any changes to the actual functionality, just changes to how the UI behaves.

**Comment by plasmacorral** [comment]
Could we please have a screenshot of the update made wrt spacing and in response to this [comment]?(https://github.com/MetaMask/metamask-extension/pull/19132#issuecomment-1548057171)

**Comment by montelaidev** [comment]
> Did this work before if you just left the password field blank? I'm mostly guessing that because I don't see any changes to the actual functionality, just changes to how the UI behaves.

Yea, users were able to pass an empty password before.

**Review by HowardBraham — APPROVED** [review]

**Comment by danroc** [comment]
@montelaidev, do you have test files (with and without a password)?

**Review by danroc — APPROVED** [review]

**Review by HowardBraham — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Pull request merged**